### PR TITLE
Make `rustc_parse_format` compile on stable again

### PR DESCRIPTION
**Pull request by fmease**
Fixes [#115948 (review comment)](https://github.com/rust-lang/rust/pull/115948/files/822233559619e7e77d984f9020d05302b784cf50#r1385932710).
cc @Veykril @notriddle

r? compiler

### CODE REVIEW

File: compiler/rustc_parse_format/src/lib.rs [hunk 1]
```diff
@@ -9,13 +9,10 @@
...
 )]
-#![cfg_attr(not(bootstrap), doc(rust_logo))]
-#![cfg_attr(not(bootstrap), allow(internal_features))]
-#![cfg_attr(not(bootstrap), feature(rustdoc_internals))]
 #![deny(rustc::untranslatable_diagnostic)]
 #![deny(rustc::diagnostic_outside_of_impl)]
-// We want to be able to build this crate with a stable compiler, so no
-// `#![feature]` attributes should be added.
+// WARNING: We want to be able to build this crate with a stable compiler,
+//          so no `#![feature]` attributes should be added!
```
Comment (fmease) [inline compiler/rustc_parse_format/src/lib.rs:15]: Someday I might open an MCP for `#![deny(rustc::features)]` or a lang FCP for `#![deny(features)]` :^)

Comment (est31) [reply compiler/rustc_parse_format/src/lib.rs:15]: I think a simple PR to add the lint would be enough for that. Lints are explicitly not stable, and this is a lint for an *unstable* feature (the feature attribute).

**Comment by Nilstrieb** [comment]
can we add a CI builder that builds those crates with the bootstrap compiler (without RUSTC_BOOTSTRAP)?

**Comment by fmease** [comment]
That's not my metier but I can read up on it.

**Comment by fmease** [comment]
Can we merge this as is? I think this unblocks the next r-a sync? I'm gonna open a t-bootstrap/t-infra ticket for the building crates which ought to compile without unstable features without `RUSTC_BOOTSTRAP=1` in CI.

**Comment by Nilstrieb** [comment]
@bors r+

feel free to r- and r=me with an even bigger and scarier warning if you want to

**Comment by fmease** [comment]
cc #117937

**Pull request merged**